harness/tests/core/test_checkpoint.py

```diff
@@ -141,3 +143,81 @@ def do_test() -> None:
                     pass
             storage_manager.restore_path.assert_called_once()
             storage_manager.restore_path.reset_mock()
+
+
+@pytest.mark.parametrize(
+    "resources,expected_merged,expected_conflicts",
+    [
+        ([{"file0": 0}, {"file1": 0}], {"file0": 0, "file1": 0}, {}),
+        ([{"file0": 0}, {"file0": 0}], {"file0": 0}, {"file0": [0, 1]}),
+        ([{"dir1/": 0}, {"dir1/": 0}], {"dir1/": 0}, {}),
+        ([{"file1/": 0}, {"file1": 0}], {"file1/": 0, "file1": 0}, {"file1": [0, 1]}),
+        ([{"dir1/file1": 0}, {"file1": 0}], {"dir1/file1": 0, "file1": 0}, {}),
+        (
+            [{"dir1/file1": 0}, {"dir1/file1/": 0}],
+            {"dir1/file1": 0, "dir1/file1/": 0},
+            {"dir1/file1": [0, 1]},
+        ),
+    ],
+)
+def test_merge_files(
+    resources: List[Dict[str, int]],
+    expected_merged: Dict[str, int],
+    expected_conflicts: Dict[str, List[int]],
+) -> None:
+    merged, conflicts = core._checkpoint.merge_resources(resources)
+    assert conflicts == expected_conflicts
+    assert merged == expected_merged
+
+
+@pytest.mark.parametrize(
+    "metadata,expected_merged,expected_conflicts",
+    [
+        ([{"a": 0}, {"b": 0}], {"a": 0, "b": 0}, {}),
+        ([{"a": 0}, {"a": 0}], {"a": 0}, {"/a": [0, 1]}),
```

Comment:
This conflict should be reported as `a` (which indicates a relative path) not `/a` (which indicates an absolute path, which this is not)

Reply:
I don't know why I had in my mind that this was reporting files in filesystems.

I no longer feel strongly about this.  It could reasonably go either way.